scripts/jenkins/Jenkinsfile

```diff
@@ -0,0 +1,515 @@
+// FIXME sloc publish errors
+// XXX add missing jobs
+// TODO have a per plugin/binding deps in Dockerfile for easier maintanence
+// TODO add warnings plugins to scan for compiler warnings
+// XXX setup way to determine what tests should run
+//           regex on env.ghprbCommentBody
+
+// # Set Variables #
+CMPVERSION = '0.8.22'
+VERSION = '0.8.23'
+
+docker_node_label = 'docker'
+docker_image_prefix = 'build-elektra-'
+
+// ## Setup log discarder ##
+properties([
+    buildDiscarder(logRotator(numToKeepStr: '60', artifactNumToKeepStr: '60'))
+])
+
+// ## Define Docker images ##
+DOCKER_IMAGES = [
+    sid: [
+        id: docker_image_prefix + 'debian-sid',
+        context: "./doc/docker/debian/sid",
+        file: "./doc/docker/debian/sid/Dockerfile"
+    ],
+    stretch: [
+        id: docker_image_prefix + 'debian-stretch',
+        context: "./doc/docker/debian/stretch",
+        file: "./doc/docker/debian/stretch/Dockerfile",
+    ],
+    stretch_minimal: [
+        id: docker_image_prefix + 'debian-stretch-minimal',
+        context: "./doc/docker/debian/stretch",
+        file: "./doc/docker/debian/stretch/Dockerfile.minimal",
+    ],
+    stretch_doc: [
+        id: docker_image_prefix + 'debian-stretch-doc',
+        context: "./doc/docker/debian/stretch",
+        file: "./doc/docker/debian/stretch/Dockerfile.doc",
+    ],
+    jessie: [
+        id: docker_image_prefix + 'debian-jessie',
+        context: "./doc/docker/debian/jessie",
+        file: "./doc/docker/debian/jessie/Dockerfile",
+    ],
+    xenial: [
+        id: docker_image_prefix + 'ubuntu-xenial',
+        context: "./doc/docker/ubuntu/xenial",
+        file: "./doc/docker/ubuntu/xenial/Dockerfile",
+    ]
+]
+
+// ## Define Build Flags ##
+BUILD_FLAGS_STANDARD = [
+    'SITE': '${STAGE_NAME}',
+    'KDB_DB_SYSTEM': '${HOME}/.config/kdb/system',
+    'KDB_DB_SPEC': '${HOME}/.config/kdb/spec',
+    'CMAKE_INSTALL_PREFIX': '${WORKSPACE}/system',
+    'INSTALL_SYSTEM_FILES': 'OFF',
+    'BUILD_DOCUMENTATION': 'OFF'
+]
+
+BUILD_FLAGS_ALL = [
+    'BINDINGS': 'ALL;EXPERIMENTAL;-DEPRECATED',
+    'PLUGINS': 'ALL;EXPERIMENTAL;-DEPRECATED',
+    'TOOLS': 'ALL'
+]
+
+BUILD_FLAGS_COVERAGE = [
+    'ENABLE_COVERAGE': 'ON'
```

Comment:
For every build job?

Reply:
Can you elaborate? It only enables the coverage when used in the test() functions later (and modifies the job accordingly so usage is recorded)

Reply:
For me some docu was missing. It was unclear if these flags are applied to every build jobs.

So simply move what you said to the docu.

Reply:
still open: please add more docu

Ideally give a reader some guide how to read the Jenkinsfile in the very top and guide through some standard use cases (e.g. how to add a build job)

Reply:
Please add a suggestion on how to archive that. I don't see comments in the c files explaining what a global variable is.

If I add comments for each bit of groovy syntax it will get unreadable very quickly.